File: python/tests/test_bowling_game.py
import unittest

import os
import sys
sys.path.append(os.path.join(os.path.dirname(__file__), '..'))
from bowling_game import bowling_game as bg

class TestBowlingGame(unittest.TestCase):
    def test_all_zeros(self):
        game = bg.BowlingGame()
        for i in range(20):
            game.roll(0)
        self.assertEqual(game.score(), 0)

    def test_all_ones(self):
        game = bg.BowlingGame()
        for i in range(20):
            game.roll(1)
        self.assertEqual(game.score(), 20)

    def test_one_spare(self):
        game = bg.BowlingGame()
        game.roll(5)
        game.roll(5)
        game.roll(3)
        for i in range(17):
            game.roll(0)
        self.assertEqual(game.score(), 16)

    def test_one_strike(self):
        game = bg.BowlingGame()
        game.roll(10)
        game.roll(3)
        game.roll(4)
        for i in range(16):
            game.roll(0)
        self.assertEqual(game.score(), 24)

    def test_perfect_game(self):
        game = bg.BowlingGame()
        for i in range(12):
            game.roll(10)
        self.assertEqual(game.score(), 300)


if __name__ == '__main__':
    unittest.main()

File: lua/bowling_game/test_bowling_game.lua
-- 必要なモジュールを読み込む
local BowlingGame = require("bowling_game")
local luaunit = require("luaunit")

-- BowlingGameクラスのユニットテスト
TestBowlingGame = {}

-- セットアップ関数（各テストケースの前に実行）
function TestBowlingGame:setUp()
    -- ボーリングゲームのインスタンスを作成
    self.game = BowlingGame:new()
end

function TestBowlingGame:testAllZeros()
    for i = 1, 20 do
        self.game:roll(0)
    end
    luaunit.assertEquals(self.game:score(), 0)
end

function TestBowlingGame:testAllOnes()
    for i = 1, 20 do
        self.game:roll(1)
    end
    luaunit.assertEquals(self.game:score(), 20)
end

function TestBowlingGame:testOneSpare()
    self.game:roll(5)
    self.game:roll(5)
    self.game:roll(3)
    for i = 1, 17 do
        self.game:roll(0)
    end
    luaunit.assertEquals(self.game:score(), 16)
end

function TestBowlingGame:testOneStrike()
    self.game:roll(10)
    self.game:roll(3)
    self.game:roll(4)
    for i = 1, 16 do
        self.game:roll(0)
    end
    luaunit.assertEquals(self.game:score(), 24)
end

function TestBowlingGame:testPerfectGame()
    for i = 1, 12 do
        self.game:roll(10)
    end
    luaunit.assertEquals(self.game:score(), 300)
end

-- テストの実行
luaunit.run()
